dnsimple/dnsimple-support
# Trigger to open the support widget

This PR makes the support widget fill more of the screen, as well:

- Adds `data-dnsimple-open-support-widget`: can be added to any DOM element to trigger the opening of the widget.
- Adds `data-dnsimple-current-site-url="https://NOTHING.dnsimple.com"`: used to determine if the article is for the site the customer is already on. For example, if the customer is @ dnsimple.com and they search for alias, we want the result to show in the support widget browser. If they are on support.dnsimple.com, then we want to do a full page load to the result.

![image](https://github.com/user-attachments/assets/919985e9-8351-499e-ae13-879aa2a9e40f)

## QA

Test using `yarn widget`:

- Search for `alias`.
- The search result links should open in the widget (because `data-dnsimple-current-site-url="https://NOTHING.dnsimple.com"` doesn't match the current site).

Test the [preview](https://deploy-preview-1331--dnsimple-support.netlify.app/):

- Search for `alias`.
- Click on a result should change the page like a normal page click (not use the support widget browser OR be in a new tab).

Test on DNSimple.com:

```js
[...document.querySelectorAll('#dnsimple-support-wrapper, #dnsimple-support-widget')].forEach((el) => el && document.body.removeChild(el))

const $script = document.createElement('script');
$script.type = 'text/javascript';
$script.src = `https://deploy-preview-1331--dnsimple-support.netlify.app/widget.js`;

document.getElementsByTagName('head')[0].appendChild($script);
```

- The search result links should open in the widget.
